apps/supernova/src/hooks/useUrlQueryState/index.ts

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useLayoutEffect } from "react"
+import { readStateFromUrl, saveStateToUrl } from "@cloudoperators/juno-url-state-provider"
+import {
+  useFilterLabels,
+  useGetStateForUrl,
+  useGlobalsActions,
+  useSilencesActions,
+  useFilterActions,
+} from "../../components/StoreProvider"
+import { transformAppStateToUrlState, transformUrlStateToAppState } from "./utils"
+
+const useUrlQueryState = () => {
+  const urlReadRef = React.useRef(false)
+  const { setActiveFilters, setActivePredefinedFilter, setSearchTerm } = useFilterActions()
+  const { setSilencesRegEx, setSilencesStatus } = useSilencesActions()
+  const { setShowDetailsFor, setActiveSelectedTab } = useGlobalsActions()
+  const filterLabels = useFilterLabels()
+  const appState = useGetStateForUrl()
+
+  // Reflect app state to the URL
+  useEffect(() => {
+    if (urlReadRef.current && urlReadRef.current) {
+      saveStateToUrl(transformAppStateToUrlState(appState))
+    }
+  }, [appState])
+
+  // Read URL state and set it to the app state this typically happens on the first render
+  useLayoutEffect(() => {
+    const stateFromTheUrl = readStateFromUrl()
+    if (Object.keys(stateFromTheUrl).length !== 0) {
+      const appState = transformUrlStateToAppState(stateFromTheUrl, filterLabels)
+      setActiveFilters(appState.activeFilters)
```

Comment:
We could compare in a smart way if the different states have changed before updating the state and generating unnecessary rerenders.

Reply:
eventually this is going to be removed.